vital/types/local_cartesian.cxx

```diff
@@ -0,0 +1,446 @@
+/*ckwg +29
+ * Copyright 2019 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the implementation of a local geographical offset coordinate system.
+ */
+
+#include <vital/types/local_cartesian.h>
+#include <vital/types/geodesy.h>
+
+#include <iomanip>
+#include <stdexcept>
+
+namespace kwiver {
+namespace vital {
+
+const double PI = 3.14159265358979323e0;
+const double PI_OVER_2 = (PI / 2.0e0);
+const double TWO_PI = (2.0 * PI);
+const double DEG_TO_RAD = (PI / 180.0);
+const double RAD_TO_DEG = (180.0 / PI);
+const double AD_C = 1.0026000;            /* Toms region 1 constant */
+const double COS_67P5 = 0.38268343236508977;  /* cosine of 67.5 degrees */
+
+/**
+ * @Brief Local Cartesian Conversion Math Utility
+ *
+ * Base on the NGA GeoTrans library
+ * https://earth-info.nga.mil/GandG/update/index.php?action=home
+ *
+ * This class is a cleaned up version of the LocalCartesian and Geocentric classes provided in GeoTrans
+ * This allows a user to define a local cartesian coordinate system with any origin (expressed in WGS84)
+ */
+class local_cartesian::geotrans
+{
+public:
+  geotrans(geo_point const& origin, double orientation)
+  {
+    // Using WGS84 ellipsoid
+    semiMajorAxis = 6378137.0;
```

Comment:
I take it this code is wholesale copied from GeoTrans, which is why the naming conventions do not conform to KWIVER norms?

TBH, I am not going to try to wade through most of this, as looking at it long enough to feel like I actually understand the math would take an inordinate amount of time.

That said, I would at least encourage looking over this for "stupid" things. For example, it appears that `semiMajorAxis` is only set here and never changes, which suggests it should instead be a `constexpr static` member. It looks like there are probably many things that could be made `constexpr static`.